### PR TITLE
Squashed commit of the following:

### PR DESCRIPTION
**Pull request by ricardoquesada**
commit aff4e27200a77db60b13ea30c2457558e5f53059
Author: Ricardo Quesada ricardoquesada@gmail.com
Date:   Thu Dec 10 17:58:41 2015 -0800

```
compiles with new GC model
```

commit 1fa69cd71231d56371cd45a378e50a1888308b42
Author: Ricardo Quesada ricardoquesada@gmail.com
Date:   Thu Dec 10 17:41:15 2015 -0800

```
Animation3D works Ok
```

commit d439969caf7e6fe83a74e37d078c4361a08cb816
Author: Ricardo Quesada ricardoquesada@gmail.com
Date:   Thu Dec 10 13:39:50 2015 -0800

```
Sprite3d create: converted to new API
```

commit aabe449e4a968fad882c44df9be787eb7c3bfcec
Author: Ricardo Quesada ricardoquesada@gmail.com
Date:   Thu Dec 10 13:27:33 2015 -0800

```
ouch...
```

commit 688ab610a8cb7607bc3c51b8ca01d800ef3c9794
Author: Ricardo Quesada ricardoquesada@gmail.com
Date:   Thu Dec 10 13:12:25 2015 -0800

```
Spawn/Sequence init* were not bound

Fixed. Now JS Scheduler Test works Ok
```

### CODE REVIEW

**Comment by zilongshanren** [comment]
@ricardoquesada 
:+1:  Finally the JSB memory model has been improved :smile: 

Should we include this improvement in v3.10 or the future release?
Now we have a separate v3.10 branch in the upstream.

File: cocos/scripting/js-bindings/manual/ScriptingCore.cpp [hunk 1]
```diff
@@ -430,8 +430,9 @@ void registerDefaultClasses(JSContext* cx, JS::HandleObject global) {
     JS_DefineFunction(cx, global, "__isObjectValid", ScriptingCore::isObjectValid, 1, JSPROP_READONLY | JSPROP_PERMANENT);
 }
 
-static void sc_finalize(JSFreeOp *freeOp, JSObject *obj) {
-    CCLOGINFO("jsbindings: finalizing JS object %p (global class)", obj);
+static void sc_finalize(JSFreeOp *freeOp, JSObject *obj)
```
Comment (pandamicro) [inline cocos/scripting/js-bindings/manual/ScriptingCore.cpp:433]: What is this function used for ?

File: tests/js-tests/src/MemoryModelTest/MemoryModelTest.js [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        var tag = 10;
+        this.addChild(sprite, 0, tag);
+        var x = winSize.width / 2;
+        var y = winSize.height / 2;
+        sprite.setPosition(x, y);
+
+        // add random property
+        sprite.randomProperty = "hello world";
+
+        sprite = this.getChildByTag(tag);
```
Comment (pandamicro) [inline tests/js-tests/src/MemoryModelTest/MemoryModelTest.js:84]: This should be placed in a scheduled callback, the sprite is still in its original scope, so won't be garbage collected

**Comment by pandamicro** [comment]
> Should we include this improvement in v3.10 or the future release?
> Now we have a separate v3.10 branch in the upstream.

@zilongshanren  It's not ready for v3.10, we need many test, I have following commits too, if all goes well, I think it can be alright to merge in v3.11

File: cocos/scripting/js-bindings/manual/ScriptingCore.cpp (hunk 1 above)
Comment (ricardoquesada) [inline cocos/scripting/js-bindings/manual/ScriptingCore.cpp:433]: I don't know... someone is using it. I just did some minor coding-style fixes

File: tests/js-tests/src/MemoryModelTest/MemoryModelTest.js (hunk 2 above)
Comment (ricardoquesada) [inline tests/js-tests/src/MemoryModelTest/MemoryModelTest.js:84]: Yes. probably we need to add that test as well. we still need to do a lot of tests.

**Comment by ricardoquesada** [comment]
@pandamicro @zilongshanren 

I think it should be part of v3.10, but it will be disabled. But advanced users can test if they want. I we can start telling them the benefits, receive feedback from them, etc.

@linshun thoughts?

**Pull request merged**